specification/cognitiveservices/data-plane/AnomalyDetector/stable/v1.1/cadl/multivariate/routes.cadl

```diff
@@ -0,0 +1,130 @@
+import "@azure-tools/cadl-azure-core";
+import "@cadl-lang/rest";
+import "./models.cadl";
+
+using Cadl.Rest;
+using Cadl.Http;
+
+namespace AnomalyDetector.Multivariate;
+
+@get
+@route("/multivariate/detect-batch/{resultId}")
+@summary("Get Multivariate Anomaly Detection Result")
+@doc("""
+For asynchronous inference, get multivariate anomaly detection result based on
+resultId returned by the BatchDetectAnomaly api.
+""")
+// To improve CADL implementation apply: https://github.com/Azure/azure-rest-api-specs/issues/21527
+op GetBatchDetectionResult(
+  @format("uuid")
+  @path resultId: string,
+): DetectionResult | ResponseError;
+
+
+@post
+@route("/multivariate/models")
+@summary("Train a Multivariate Anomaly Detection Model")
+@doc("""
+Create and train a multivariate anomaly detection model. The request must
+include a source parameter to indicate an externally accessible Azure blob
+storage URI.There are two types of data input: An URI pointed to an Azure blob
+storage folder which contains multiple CSV files, and each CSV file contains
+two columns, timestamp and variable. Another type of input is an URI pointed to
+a CSV file in Azure blob storage, which contains all the variables and a
+timestamp column.
+""")
+// To improve CADL implementation apply: https://github.com/Azure/azure-rest-api-specs/issues/21527
+op CreateAndTrainMultivariateModel(
+  @body body: ModelInfo,
+): {
+  @statusCode statusCode: 201,
+  @doc("Location and ID of the model.")
+  @header Location: string,
+  @body body: Model
+  } | ResponseError;
```

Comment:
This is matching what is currently defined in the swagger. @jr-MS is this the right definition?

Reply:
We can leave it at SDK archboard.

Yes, it is same as swagger and your backend. Just a precautions, as the API in SDK won't be e.g. `Model createAndTrainMultivariateModel(Model model)`.

But given it is `createAndTrain`, it might be fine.